internal/cmd/log-file-metric-exporter/main.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"flag"
+	"github.com/ViaQ/logerr/log"
+	"os"
+	"path"
+	"strings"
+	"net/http"
+	"github.com/openshift/cluster-logging-operator/internal/pkg/symnotify"
+        "github.com/prometheus/client_golang/prometheus"
+        "github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+
+
+
+var (
+ verbosity int = 0
+ namespace string = "unknown"
+ podname string = "unknown"
+ containername string ="unknown"
+)
+
+
+
+type FileWatcher struct {
+        watcher *symnotify.Watcher
+        metrics *prometheus.CounterVec
+        sizes  map[string]float64
+        added  map[string]bool
+}
+
+
+func (w *FileWatcher) Update(path string, namespace string, podname string, containername string) error {
+        counter, err := w.metrics.GetMetricWithLabelValues(path,namespace,podname,containername)
+        if err != nil {
+                return err
+        }
+        stat, err := os.Stat(path)
+        if err != nil {
+                return err
+        }
+        if stat.IsDir() {
+                return nil // Ignore directories
+        }
+        lastSize, size := float64(w.sizes[path]), float64(stat.Size())
+        w.sizes[path] = size
+        var add float64
+        if size > lastSize {
+                // File has grown, add the difference to the counter.
+                add = size - lastSize
+        } else if size < lastSize {
+                // File truncated, starting over. Add the size.
+                add = size
+        }
+        log.V(2).Info("For logfile in...", "path",path,"lastsize",lastSize,"currentsize",size,"addedbytes",add)
+	counter.Add(add)
+        return nil
+}
+
+func (w *FileWatcher) Watch( ) {
+
+        
+	for {
+		//All logfiles with containername are added to the watcher
+		//write event for these logfiles are being watched
+		//create event gets issued for all new logfiles appear under logfilepathname /var/log/containers/
+		//For the cases new log files added, old files moved, old files deleted, you need to add/remove them from watcher as whole dir added to the watcher
+		//For new log files added write event is not getting issued
+		
+                e, err := w.watcher.Event()
+                if err != nil {
+		log.Error(err, "Watcher.Event returning err")
+		os.Exit(1)
+		}
+
+                log.V(2).Info("Events notified for...", "e.Name",e.Name,"Event",e.Op)
+
+		//Get namespace, podname, containername from e.Name - log file path
```

Comment:
We should be able to do this with a compiled regex.   ref: https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter/blob/master/lib/fluent/plugin/filter_kubernetes_metadata.rb#L56

Reply:
@jcantrill  are you suggesting equivalent regex based parsing to be written in Golang ?

Reply:
yes. instead of string manipulation